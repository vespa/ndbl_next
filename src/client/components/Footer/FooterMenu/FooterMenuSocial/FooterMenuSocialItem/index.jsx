
import React from 'react';
import PropTypes from 'prop-types';
import cssMain from '../../../../../../scss/main.scss';

const FooterMenuSocialSocialItem = ({ text, icon, href }) => {
  const classe = `icon__social--${icon}`;
  return (
    <li>
      <span className={`${cssMain[classe]} ${cssMain.icon__social} ${cssMain.icon}`}> {text} </span>
      <a href={href}> {text} </a>
    </li>
  );
};

FooterMenuSocialSocialItem.propTypes = {
  text: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  href: PropTypes.string.isRequired,
};

export default FooterMenuSocialSocialItem;
