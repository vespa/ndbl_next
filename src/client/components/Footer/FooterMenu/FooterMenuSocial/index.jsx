
import React from 'react';
import PropTypes from 'prop-types';
import css from './style.scss';
import cssMain from '../../../../../scss/main.scss';
import FooterMenuSocialItem from './FooterMenuSocialItem';

const FooterMenuSocial = ({ title }) => {
  return (
    <React.Fragment>
      <h3 className={css['footer-menu__title']}>{title}</h3>
      <div className={css['footer-menu__list']}>
        <ul className={`${cssMain.col_6} ${css['footer-menu__list']}}`}>
          {[
            { text: 'Newsletter', icon: 'newsletter', href: '/' },
            { text: 'Instagram', icon: 'instagram', href: '/' },
            { text: 'Twitter', icon: 'twitter', href: '/' },
            { text: 'Tumblr', icon: 'tumblr', href: '/' },
            { text: 'Linkedin', icon: 'linkedin', href: '/' },
          ].map(item => <FooterMenuSocialItem key={item.text} {...item} />)
          }
        </ul>
        <ul className={`${cssMain.col_6} ${css['footer-menu__list']}}`}>
          {[
            { text: 'Facebook', icon: 'facebook', href: '/' },
            { text: 'Pinterest', icon: 'pinterest', href: '/' },
            { text: 'Youtube', icon: 'youtube', href: '/' },
            { text: 'Google+', icon: 'google', href: '/' },
          ].map(item => <FooterMenuSocialItem key={item.text} {...item} />)
          }
        </ul>
      </div>
    </React.Fragment>
  );
};

FooterMenuSocial.propTypes = {
  title: PropTypes.string.isRequired,
};

export default FooterMenuSocial;
