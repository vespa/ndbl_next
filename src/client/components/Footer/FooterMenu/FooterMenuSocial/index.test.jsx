import React from 'react';
import { shallow } from 'enzyme';
import FooterMenuSocial from './index';

describe('[Component: FooterMenuSocial]', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<FooterMenuSocial title="test" />);
  });
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
