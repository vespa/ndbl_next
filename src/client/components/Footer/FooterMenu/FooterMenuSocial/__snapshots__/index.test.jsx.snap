// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component: FooterMenuSocial] should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FooterMenuSocial
    title="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h3
          className={undefined}
        >
          test
        </h3>,
        <div
          className={undefined}
        >
          <ul
            className="undefined undefined}"
          >
            <FooterMenuSocialSocialItem
              href="/"
              icon="newsletter"
              text="Newsletter"
            />
            <FooterMenuSocialSocialItem
              href="/"
              icon="instagram"
              text="Instagram"
            />
            <FooterMenuSocialSocialItem
              href="/"
              icon="twitter"
              text="Twitter"
            />
            <FooterMenuSocialSocialItem
              href="/"
              icon="tumblr"
              text="Tumblr"
            />
            <FooterMenuSocialSocialItem
              href="/"
              icon="linkedin"
              text="Linkedin"
            />
          </ul>
          <ul
            className="undefined undefined}"
          >
            <FooterMenuSocialSocialItem
              href="/"
              icon="facebook"
              text="Facebook"
            />
            <FooterMenuSocialSocialItem
              href="/"
              icon="pinterest"
              text="Pinterest"
            />
            <FooterMenuSocialSocialItem
              href="/"
              icon="youtube"
              text="Youtube"
            />
            <FooterMenuSocialSocialItem
              href="/"
              icon="google"
              text="Google+"
            />
          </ul>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
          "className": undefined,
        },
        "ref": null,
        "rendered": "test",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ul
              className="undefined undefined}"
            >
              <FooterMenuSocialSocialItem
                href="/"
                icon="newsletter"
                text="Newsletter"
              />
              <FooterMenuSocialSocialItem
                href="/"
                icon="instagram"
                text="Instagram"
              />
              <FooterMenuSocialSocialItem
                href="/"
                icon="twitter"
                text="Twitter"
              />
              <FooterMenuSocialSocialItem
                href="/"
                icon="tumblr"
                text="Tumblr"
              />
              <FooterMenuSocialSocialItem
                href="/"
                icon="linkedin"
                text="Linkedin"
              />
            </ul>,
            <ul
              className="undefined undefined}"
            >
              <FooterMenuSocialSocialItem
                href="/"
                icon="facebook"
                text="Facebook"
              />
              <FooterMenuSocialSocialItem
                href="/"
                icon="pinterest"
                text="Pinterest"
              />
              <FooterMenuSocialSocialItem
                href="/"
                icon="youtube"
                text="Youtube"
              />
              <FooterMenuSocialSocialItem
                href="/"
                icon="google"
                text="Google+"
              />
            </ul>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <FooterMenuSocialSocialItem
                  href="/"
                  icon="newsletter"
                  text="Newsletter"
                />,
                <FooterMenuSocialSocialItem
                  href="/"
                  icon="instagram"
                  text="Instagram"
                />,
                <FooterMenuSocialSocialItem
                  href="/"
                  icon="twitter"
                  text="Twitter"
                />,
                <FooterMenuSocialSocialItem
                  href="/"
                  icon="tumblr"
                  text="Tumblr"
                />,
                <FooterMenuSocialSocialItem
                  href="/"
                  icon="linkedin"
                  text="Linkedin"
                />,
              ],
              "className": "undefined undefined}",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "Newsletter",
                "nodeType": "function",
                "props": Object {
                  "href": "/",
                  "icon": "newsletter",
                  "text": "Newsletter",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "Instagram",
                "nodeType": "function",
                "props": Object {
                  "href": "/",
                  "icon": "instagram",
                  "text": "Instagram",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "Twitter",
                "nodeType": "function",
                "props": Object {
                  "href": "/",
                  "icon": "twitter",
                  "text": "Twitter",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "Tumblr",
                "nodeType": "function",
                "props": Object {
                  "href": "/",
                  "icon": "tumblr",
                  "text": "Tumblr",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "Linkedin",
                "nodeType": "function",
                "props": Object {
                  "href": "/",
                  "icon": "linkedin",
                  "text": "Linkedin",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "ul",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <FooterMenuSocialSocialItem
                  href="/"
                  icon="facebook"
                  text="Facebook"
                />,
                <FooterMenuSocialSocialItem
                  href="/"
                  icon="pinterest"
                  text="Pinterest"
                />,
                <FooterMenuSocialSocialItem
                  href="/"
                  icon="youtube"
                  text="Youtube"
                />,
                <FooterMenuSocialSocialItem
                  href="/"
                  icon="google"
                  text="Google+"
                />,
              ],
              "className": "undefined undefined}",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "Facebook",
                "nodeType": "function",
                "props": Object {
                  "href": "/",
                  "icon": "facebook",
                  "text": "Facebook",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "Pinterest",
                "nodeType": "function",
                "props": Object {
                  "href": "/",
                  "icon": "pinterest",
                  "text": "Pinterest",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "Youtube",
                "nodeType": "function",
                "props": Object {
                  "href": "/",
                  "icon": "youtube",
                  "text": "Youtube",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "Google+",
                "nodeType": "function",
                "props": Object {
                  "href": "/",
                  "icon": "google",
                  "text": "Google+",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "ul",
          },
        ],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h3
            className={undefined}
          >
            test
          </h3>,
          <div
            className={undefined}
          >
            <ul
              className="undefined undefined}"
            >
              <FooterMenuSocialSocialItem
                href="/"
                icon="newsletter"
                text="Newsletter"
              />
              <FooterMenuSocialSocialItem
                href="/"
                icon="instagram"
                text="Instagram"
              />
              <FooterMenuSocialSocialItem
                href="/"
                icon="twitter"
                text="Twitter"
              />
              <FooterMenuSocialSocialItem
                href="/"
                icon="tumblr"
                text="Tumblr"
              />
              <FooterMenuSocialSocialItem
                href="/"
                icon="linkedin"
                text="Linkedin"
              />
            </ul>
            <ul
              className="undefined undefined}"
            >
              <FooterMenuSocialSocialItem
                href="/"
                icon="facebook"
                text="Facebook"
              />
              <FooterMenuSocialSocialItem
                href="/"
                icon="pinterest"
                text="Pinterest"
              />
              <FooterMenuSocialSocialItem
                href="/"
                icon="youtube"
                text="Youtube"
              />
              <FooterMenuSocialSocialItem
                href="/"
                icon="google"
                text="Google+"
              />
            </ul>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test",
            "className": undefined,
          },
          "ref": null,
          "rendered": "test",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ul
                className="undefined undefined}"
              >
                <FooterMenuSocialSocialItem
                  href="/"
                  icon="newsletter"
                  text="Newsletter"
                />
                <FooterMenuSocialSocialItem
                  href="/"
                  icon="instagram"
                  text="Instagram"
                />
                <FooterMenuSocialSocialItem
                  href="/"
                  icon="twitter"
                  text="Twitter"
                />
                <FooterMenuSocialSocialItem
                  href="/"
                  icon="tumblr"
                  text="Tumblr"
                />
                <FooterMenuSocialSocialItem
                  href="/"
                  icon="linkedin"
                  text="Linkedin"
                />
              </ul>,
              <ul
                className="undefined undefined}"
              >
                <FooterMenuSocialSocialItem
                  href="/"
                  icon="facebook"
                  text="Facebook"
                />
                <FooterMenuSocialSocialItem
                  href="/"
                  icon="pinterest"
                  text="Pinterest"
                />
                <FooterMenuSocialSocialItem
                  href="/"
                  icon="youtube"
                  text="Youtube"
                />
                <FooterMenuSocialSocialItem
                  href="/"
                  icon="google"
                  text="Google+"
                />
              </ul>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <FooterMenuSocialSocialItem
                    href="/"
                    icon="newsletter"
                    text="Newsletter"
                  />,
                  <FooterMenuSocialSocialItem
                    href="/"
                    icon="instagram"
                    text="Instagram"
                  />,
                  <FooterMenuSocialSocialItem
                    href="/"
                    icon="twitter"
                    text="Twitter"
                  />,
                  <FooterMenuSocialSocialItem
                    href="/"
                    icon="tumblr"
                    text="Tumblr"
                  />,
                  <FooterMenuSocialSocialItem
                    href="/"
                    icon="linkedin"
                    text="Linkedin"
                  />,
                ],
                "className": "undefined undefined}",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "Newsletter",
                  "nodeType": "function",
                  "props": Object {
                    "href": "/",
                    "icon": "newsletter",
                    "text": "Newsletter",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "Instagram",
                  "nodeType": "function",
                  "props": Object {
                    "href": "/",
                    "icon": "instagram",
                    "text": "Instagram",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "Twitter",
                  "nodeType": "function",
                  "props": Object {
                    "href": "/",
                    "icon": "twitter",
                    "text": "Twitter",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "Tumblr",
                  "nodeType": "function",
                  "props": Object {
                    "href": "/",
                    "icon": "tumblr",
                    "text": "Tumblr",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "Linkedin",
                  "nodeType": "function",
                  "props": Object {
                    "href": "/",
                    "icon": "linkedin",
                    "text": "Linkedin",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "ul",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <FooterMenuSocialSocialItem
                    href="/"
                    icon="facebook"
                    text="Facebook"
                  />,
                  <FooterMenuSocialSocialItem
                    href="/"
                    icon="pinterest"
                    text="Pinterest"
                  />,
                  <FooterMenuSocialSocialItem
                    href="/"
                    icon="youtube"
                    text="Youtube"
                  />,
                  <FooterMenuSocialSocialItem
                    href="/"
                    icon="google"
                    text="Google+"
                  />,
                ],
                "className": "undefined undefined}",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "Facebook",
                  "nodeType": "function",
                  "props": Object {
                    "href": "/",
                    "icon": "facebook",
                    "text": "Facebook",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "Pinterest",
                  "nodeType": "function",
                  "props": Object {
                    "href": "/",
                    "icon": "pinterest",
                    "text": "Pinterest",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "Youtube",
                  "nodeType": "function",
                  "props": Object {
                    "href": "/",
                    "icon": "youtube",
                    "text": "Youtube",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "Google+",
                  "nodeType": "function",
                  "props": Object {
                    "href": "/",
                    "icon": "google",
                    "text": "Google+",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "ul",
            },
          ],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
